Sources/web3swift/Transaction/TransactionOptions.swift

```diff
@@ -0,0 +1,286 @@
+        opts.to = mergeIfNotNil(first: self.to, second: other.to)
+        opts.from = mergeIfNotNil(first: self.from, second: other.from)
+        opts.gasLimit = mergeIfNotNil(first: self.gasLimit, second: other.gasLimit)
+        opts.gasPrice = mergeIfNotNil(first: self.gasPrice, second: other.gasPrice)
+        opts.maxFeePerGas = mergeIfNotNil(first: self.maxFeePerGas, second: other.maxFeePerGas)
+        opts.maxPriorityFeePerGas = mergeIfNotNil(first: self.maxPriorityFeePerGas, second: other.maxPriorityFeePerGas)
+        opts.value = mergeIfNotNil(first: self.value, second: other.value)
+        opts.nonce = mergeIfNotNil(first: self.nonce, second: other.nonce)
+        opts.callOnBlock = mergeIfNotNil(first: self.callOnBlock, second: other.callOnBlock)
+        return opts
+    }
+
+    /// Merges two sets of options by overriding the parameters from the first set by parameters from the second
+    /// set if those are not nil.
+    ///
+    /// Returns default option if both parameters are nil.
+    public static func merge(_ options: TransactionOptions?, with other: TransactionOptions?) -> TransactionOptions? {
+        var newOptions = TransactionOptions.defaultOptions // default has lowest priority
+        newOptions = newOptions.merge(options)
+        newOptions = newOptions.merge(other) // other has highest priority
+        return newOptions
+    }
+}
+
+private func mergeIfNotNil<T>(first: T?, second: T?) -> T? {
+    if second != nil {
+        return second
+    } else if first != nil {
+        return first
+    }
+    return nil
+}
+
+extension TransactionOptions: Decodable {
+    enum CodingKeys: String, CodingKey {
+        case type
+        case to
+        case from
+        case gasPrice
+        case gas
+        case maxFeePerGas
+        case maxPriorityFeePerGas
+        case value
+        case nonce
+        case callOnBlock
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let defaultOptions = TransactionOptions.defaultOptions
+
+        // type is guaranteed to be set after this
+        if let typeUInt = try? container.decodeHex(to: UInt.self, key: .type) {
+            if typeUInt < TransactionType.total.rawValue {
+                guard let type = TransactionType(rawValue: typeUInt) else { throw Web3Error.dataError }
+                self.type = type
+            } else { throw Web3Error.dataError }
+        } else { self.type = .legacy } // legacy streams may not have type set
+
+        let toString = try? container.decode(String.self, forKey: .to)
+        switch toString {
+        case nil, "0x", "0x0":
+            self.to = EthereumAddress.contractDeploymentAddress()
+        default:
+            guard let addressString = toString else { throw Web3Error.dataError }
+            guard let ethAddr = EthereumAddress(addressString) else { throw Web3Error.dataError }
```

Comment:
`guard let addressString = toString else { throw Web3Error.dataError }`
can be removed and force unwrapped as `nil` case is handled above.

Reply:
missed it in this file, but caught it in other places... thanks.